gluecodium/src/test/resources/smoke/listeners/output/dart/ffi/ffi_smoke_CalculatorListener.cpp

```diff
@@ -0,0 +1,162 @@
+#include "ffi_smoke_CalculatorListener.h"
+#include "ConversionBase.h"
+#include "gluecodium/VectorHash.h"
+#include "smoke/CalculationResult.h"
+#include "smoke/CalculatorListener.h"
+#include <memory>
+#include <vector>
+#include <memory>
+#include <new>
+class smoke_CalculatorListener_Proxy : public ::smoke::CalculatorListener {
+public:
+    smoke_CalculatorListener_Proxy(uint64_t token, FfiOpaqueHandle f0, FfiOpaqueHandle f1, FfiOpaqueHandle f2, FfiOpaqueHandle f3, FfiOpaqueHandle f4, FfiOpaqueHandle f5)
+        : token(token), f0(f0), f1(f1), f2(f2), f3(f3), f4(f4), f5(f5) { }
+    void
+    on_calculation_result(const double calculationResult) override {
+        int64_t _error = (*reinterpret_cast<int64_t (*)(uint64_t, double)>(f0))(token,
```

Comment:
I would suppress warning about unused variable

Reply:
That's not particularly important as it will get used in the next implementation.